lib/src/syn/v2/lexer/byte.rs

```diff
@@ -0,0 +1,388 @@
+use crate::syn::v2::{
+	lexer::{
+		unicode::{byte, chars},
+		Error, Lexer,
+	},
+	token::{t, Token, TokenKind},
+};
+
+impl<'a> Lexer<'a> {
+	/// Eats a single line comment and returns the next token.
+	pub fn eat_single_line_comment(&mut self) {
+		loop {
+			let Some(byte) = self.reader.next() else {
+				break;
+			};
+			match byte {
+				byte::CR => {
+					self.reader.peek();
+					self.eat(byte::LF);
+					break;
+				}
+				byte::LF => {
+					break;
+				}
+				x if !x.is_ascii() => {
+					// -1 because we already ate the byte.
+					let backup = self.reader.offset() - 1;
+					let char = match self.reader.complete_char(x) {
+						Ok(x) => x,
+						Err(_) => {
+							// let the next token handle the error.
+							self.reader.backup(backup);
+							break;
+						}
+					};
+
+					match char {
+						chars::LS | chars::PS => break,
+						_ => {}
+					}
+				}
+				_ => {}
+			}
+		}
+		self.set_whitespace_span(self.current_span());
+		self.skip_offset();
+	}
+
+	/// Eats a multi line comment and returns the next token.
+	pub fn eat_multi_line_comment(&mut self) -> Result<(), Error> {
+		loop {
+			let Some(byte) = self.reader.next() else {
+				return Err(Error::UnexpectedEof);
+			};
+			if let b'*' = byte {
+				let Some(byte) = self.reader.next() else {
+					return Err(Error::UnexpectedEof);
+				};
+				if b'/' == byte {
+					self.set_whitespace_span(self.current_span());
+					self.skip_offset();
+					return Ok(());
+				}
+			}
+		}
+	}
+
+	/// Eat whitespace like spaces tables and new-lines.
+	pub fn eat_whitespace(&mut self) {
+		loop {
+			let Some(byte) = self.reader.peek() else {
+				return;
+			};
+			match byte {
+				byte::CR | byte::FF | byte::LF | byte::SP | byte::VT | byte::TAB => {
+					self.reader.next();
+				}
+				x if !x.is_ascii() => {
+					let backup = self.reader.offset();
+					self.reader.next();
+					let char = match self.reader.complete_char(x) {
+						Ok(x) => x,
+						Err(_) => {
+							self.reader.backup(backup);
+							break;
+						}
+					};
+
+					match char {
```

Comment:
I did it this way because I wasn't sure that `is_whitespace` isn't more broad than this match case and since we previously already asserted that the character isn't ascii it might be slightly slower to match ascii whitespace again.